android/testing.md

```diff
@@ -0,0 +1,146 @@
+# Testing
+*This document is intended to explain which testing styles and technologies the Android teams at Mozilla use and why. It should be accessible to developers new to testing on Android.*
+
+*Presently this document describes testing on Firefox for Fire TV: it should grow to include other applications.*
+
+The primary problem we try to solve with automated testing is preventing unintended changes, which often lead to bugs or poor user experiences. This problem is most common -- thus we benefit the most from thorough testing -- when developers are modifying complex or unfamiliar code.
+
+## Quick primer on automated testing
+Automated tests are typically broken up into the following categories:
+- End-to-End tests
+  - Black box tests that act like users: the system, they click the screen and assert what is displayed, typically following common user scenarios
+  - Notable for being **the only test of the full system.** Tend to **run slowly**, **be fragile**, and **be difficult to debug**
+- Integration tests
+  - Tests comprising of multiple parts of the system: they can modify and assert internal state
+  - Compromise between end-to-end and unit tests. Tend to **run fairly quickly**
```

Comment:
Clarify: why is this fast, while e2e tests are slow? For a new developer, does this feel like "obviously we should just write integration tests"?